certbot/certbot/compat/filesystem.py

```diff
@@ -383,8 +385,30 @@ def realpath(file_path):
     return os.path.abspath(file_path)
 
 
+def readlink(link_path):
+    # type: (str) -> str
+    """
+    Return a string representing the path to which the symbolic link points.
+
+    :param str link_path: The symlink path to resolve
+    :return: The path the symlink points to
+    :returns: str
+    :raise: ValueError if a long path (260> characters) is encountered on Windows
+    """
+    path = os.readlink(link_path)
+
+    # Max length of a normal path is 260 characters on Windows, including the non printable
+    # termination character "<NUL>". The termination character is not included in Python
+    # strings, giving a max length of 259 characters, + 4 characters for the extended form
+    # prefix, to an effective max length of the studied path of 263 characters.
+    if not POSIX_MODE and path.startswith('\\\\?\\') and len(path) < 264:
+        raise ValueError("Long paths are not supported by Certbot on Windows.")
```

Comment:
I think we only want to error here if the path is long it actually requires the `\\?\` prefix. I'd expect tests on this code to fail on Python 3.8+ on Windows.

I also personally think adding a simple unit test or two for this function is a good idea.

Reply:
Sorry my implementation is wrong, I could not get the time to continue the work until now.